### PR TITLE
Add suggestion for `vue/require-name-property` rule

### PR DESCRIPTION
**Pull request by 0xYootou**
this is a pull request for issues: #2033

### CODE REVIEW

File: lib/rules/require-name-property.js [hunk 1]
```diff
@@ -27,7 +28,8 @@ module.exports = {
       categories: undefined,
       url: 'https://eslint.vuejs.org/rules/require-name-property.html'
     },
-    fixable: null,
+    fixable: 'whitespace',
+    hasSuggestions: true,
```
Comment (FloEdelmann) [inline lib/rules/require-name-property.js:32]: You added a fixer, not suggestions.

File: lib/rules/require-name-property.js [hunk 2]
```diff
@@ -47,7 +49,23 @@ module.exports = {
...
-        message: 'Required name property is not set.'
+        message: 'Required name property is not set.',
+        fix: (fixer) => {
+          const extension = path.extname(context.getFilename())
+          const filename = path.basename(context.getFilename(), extension)
+          // fix only when property is not empty
+          if (component.properties.length > 0) {
+            const startColumn = component.properties[0].loc.start.column
+            // insert name property before the first property
+            return fixer.insertTextBefore(
```
Comment (FloEdelmann) [inline lib/rules/require-name-property.js:60]: Can you please add test cases for the fixer?

**Review by FloEdelmann — CHANGES_REQUESTED** [review]
Thanks for the suggestion and the implementation!

Can we always auto-fix this to the filename or should we rather only add a suggestion for that, which the user can manually apply?

Also, please see the issues below.

**Comment by 0xYootou** [comment]
ok, i think add a suggestion for that is better too, i will rewrite the code and add test case.

**Comment by 0xYootou** [comment]
i push new code for this feature, and add some new test case for this rule and suggestions.

**Review by FloEdelmann — CHANGES_REQUESTED** [review]
Thanks for the changes!

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**